### ISSUE TITLE
Monaco editor display and cursor state bug on chromium-based browsers

### ISSUE THREAD

**Issue by Kaede25** [SYP-AHIF-2023-24-25/SolarDoc#146]
The text written in the editor doesn't get displayed correctly. The newly written text doesn't get placed where it is written, but instead at the end of the line also some of the letters do not get saved in the local storage.

![image](https://github.com/SYP-AHIF-2023-24-25/SolarDoc/assets/86598092/edee3b86-f6cd-4ac9-8e51-af9035de6bb8)

**Comment by Luna-Klatzer** [comment]
Verified to be only present in the `/editor` version of the monaco editor, `/test-editor` is unaffected.

**Comment by Luna-Klatzer** [comment]
Not reproducible as of 7b959adb4cba0f6e82379060bd6bef9c3a6272d0, potentially already fixed.

Tested on Arch Linux with Google Chrome Version 120.0.6099.216 (Official Build) (64-bit).

**Comment by Luna-Klatzer** [comment]
Also not reproducible on the same browser on Arch Linux. Perhaps a weird version or operating system-specific bug?

**Comment by Luna-Klatzer** [comment]
Reproducible on Windows Chrome Version 125.0.6422.142 (Official Build) (64-bit).

**Comment by Luna-Klatzer** [comment]
Sadly still reproducible on Windows Chrome Version 125.0.6422.142 (Official Build) (64-bit) with 7b959adb4cba0f6e82379060bd6bef9c3a6272d0. Needs further testing!

**Comment by Luna-Klatzer** [comment]
This bug fix will be applied to both `0.7.x` (released with `0.7.0`) and to `0.6.x` (causing it to bump to `0.6.2`).

**Comment by Luna-Klatzer** [comment]
Found the root cause of the bug. It was a bug, where the custom font we were using was loaded after the Monaco render itself, so the Monaco editor most likely defaulted to another font which then had a different size and scaling. Once the font loads we correctly see the font we specified, but the Monaco editor is stuck with incorrect font sizing and as such incorrect cursor states. Due to this the user always saw an incorrectly sized editor content when they first opened up the editor, as Monaco will only adjust after one or more reloads of the page.

Fixed with 7bb71490497d1bd432ef6439b672ca2c44bcb926 for `0.7.0-dev` and partially fixed by a21efd01db199209691a27fa9b0fc671ee575b79 (The full bug fix will only be present in `0.7.0`, as the local storage bug required a full restructuring of the internal editor logic -> See #150)
